File: rb/features.rb
# SearchNotice SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module SearchNoticeFeatures
  def self.make_feature(name)
    case name
    when "base"
      SearchNoticeBaseFeature.new
    when "test"
      SearchNoticeTestFeature.new
    else
      SearchNoticeBaseFeature.new
    end
  end
end

File: py/features.py
# SearchNotice SDK feature factory

from feature.base_feature import SearchNoticeBaseFeature
from feature.test_feature import SearchNoticeTestFeature


def _make_feature(name):
    features = {
        "base": lambda: SearchNoticeBaseFeature(),
        "test": lambda: SearchNoticeTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
